src/stdpar/Framework/CountingIterator.h

```diff
@@ -0,0 +1,69 @@
+struct counting_iterator {
```

Comment:
I had the impression the use of C++20 for `nvc++` would have provided the `counting_iterator` from the standard library. Did I understand wrong?

Reply:
You are correct, we can use [`std::views::iota`](https://en.cppreference.com/w/cpp/ranges/iota_view). I updated stdpar in 6553e97 but kept the iterator around, I'll remove it